src/test/java/uk/gov/hmcts/reform/unspec/service/WorkingDayIndicatorTest.java

```diff
@@ -0,0 +1,198 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+class WorkingDayIndicatorTest {
+
+    private static final LocalDate BANK_HOLIDAY = LocalDate.of(2017, 5, 29);
+    private static final LocalDate NEXT_WORKING_DAY_AFTER_BANK_HOLIDAY = LocalDate.of(2017, 5, 30);
+    private static final LocalDate PREVIOUS_WORKING_DAY_BEFORE_BANK_HOLIDAY = LocalDate.of(2017, 5, 26);
+    private static final LocalDate SATURDAY_WEEK_BEFORE = LocalDate.of(2017, 6, 3);
+    private static final LocalDate SUNDAY_WEEK_BEFORE = LocalDate.of(2017, 6, 4);
+    private static final LocalDate MONDAY = LocalDate.of(2017, 6, 5);
+    private static final LocalDate TUESDAY = LocalDate.of(2017, 6, 6);
+    private static final LocalDate WEDNESDAY = LocalDate.of(2017, 6, 7);
+    private static final LocalDate THURSDAY = LocalDate.of(2017, 6, 8);
+    private static final LocalDate FRIDAY = LocalDate.of(2017, 6, 9);
+    private static final LocalDate SATURDAY = LocalDate.of(2017, 6, 10);
+    private static final LocalDate SUNDAY = LocalDate.of(2017, 6, 11);
+
+    static class WeekDays implements ArgumentsProvider {
+
+        @Override
+        public Stream<Arguments> provideArguments(ExtensionContext context) {
+            return Stream.of(
+                Arguments.of(MONDAY),
+                Arguments.of(TUESDAY),
+                Arguments.of(WEDNESDAY),
+                Arguments.of(THURSDAY),
+                Arguments.of(FRIDAY)
+            );
+        }
+    }
+
+    static class Weekend implements ArgumentsProvider {
+
+        @Override
+        public Stream<Arguments> provideArguments(ExtensionContext context) {
+            return Stream.of(
+                Arguments.of(SATURDAY),
+                Arguments.of(SUNDAY)
+            );
+        }
+    }
+
+    private WorkingDayIndicator service;
+
+    @Mock
+    private PublicHolidaysCollection publicHolidaysApiClient;
+    @Mock
+    private NonWorkingDaysCollection nonWorkingDaysCollection;
+
+    @BeforeEach
+    void setup() {
+        service = new WorkingDayIndicator(publicHolidaysApiClient, nonWorkingDaysCollection);
+    }
+
+    @Nested
+    class ForWeekdays {
+
+        @ParameterizedTest
+        @ArgumentsSource(WeekDays.class)
+        void shouldReturnTrue_whenWeekday(LocalDate weekday) {
+            when(publicHolidaysApiClient.getPublicHolidays()).thenReturn(Collections.emptySet());
+
+            assertTrue(service.isWorkingDay(weekday));
+        }
+    }
+
+    @Nested
+    class ForWeekend {
+
+        @ParameterizedTest
+        @ArgumentsSource(Weekend.class)
+        void shouldReturnFalse_whenWeekend(LocalDate weekend) {
+            assertFalse(service.isWorkingDay(weekend));
+        }
+    }
+
+    @Nested
+    class IsWorkingDay {
+
+        @Test
+        void shouldReturnFalseForOneBankHoliday_whenThereIsOneBankHolidayInCollection() {
+            LocalDate bankHoliday = BANK_HOLIDAY;
```

Comment:
why not use BANK_HOLIDAY directly?

Reply:
yes we can but was separating concerns based on [Arrange-Act-Assert](https://github.com/testdouble/contributing-tests/wiki/Arrange-Act-Assert)  steps.